src/main/java/bern/functionalities/Ui.java

```diff
@@ -0,0 +1,276 @@
+        }
+        if (found.size() == 0) {
+            result = "There is no matching task.";
+        }
+
+        return result;
+
+    }
+
+    /**
+     * A method to get the reply if input starts with done.
+     *
+     * @param input The given command.
+     * @param arListTask The initial ArrayList of Tasks.
+     * @return The reply by the bot.
+     * @throws BernException If input is invalid.
+     */
+    public String ifDone(String input, ArrayList<Task> arListTask) throws BernException {
+        if (input.length() == 4 || (input.length() == 5 && input.substring(4, 5).equals(" "))) {
+            throw new EmptyDescriptionException("done");
+        } else if (!new Parser().isANumber(input.substring(5))) {
+            throw new InvalidCommandException(input);
+        } else if (Integer.parseInt(input.substring(5)) > arListTask.size()) {
+            throw new IndexException(input);
+        }
+        int index = Integer.parseInt(input.substring(5)) - 1;
+        arListTask.get(index).markAsDone();
+        return "Good job! I've marked this task as done:\n"
+                + arListTask.get(index).toString();
+    }
+
+    /**
+     * A method to process input from parsing to replying to writing into file.
+     *
+     * @param input The given command.
+     * @param arListTask The initial ArrayList of Tasks.
+     */
+    public void processInput(String input, ArrayList<Task> arListTask) {
+        try {
+            if (parser.isDone(input)) {
+                System.out.println(new Ui().getReply(Command.DONE, input, arListTask));
+                storage.writeIntoFile(arListTask);
+            } else if (parser.isDeadline(input)) {
+                System.out.println(new Ui().getReply(Command.DEADLINE, input, arListTask));
+                storage.writeIntoFile(arListTask);
+            } else if (parser.isEvent(input)) {
+                System.out.println(new Ui().getReply(Command.EVENT, input, arListTask));
+                storage.writeIntoFile(arListTask);
+            } else if (parser.isToDo(input)) {
+                System.out.println(new Ui().getReply(Command.TODO, input, arListTask));
+                storage.writeIntoFile(arListTask);
+            } else if (parser.isBye(input)) {
+                System.out.println(new Ui().getReply(Command.BYE, input, arListTask));
+            } else if (parser.isList(input)) {
+                System.out.println(new Ui().getReply(Command.LIST, input, arListTask));
+                storage.writeIntoFile(arListTask);
+            } else if (parser.isDelete(input)) {
+                System.out.println(new Ui().getReply(Command.DELETE, input, arListTask));
+                storage.writeIntoFile(arListTask);
+            } else if (parser.isFind(input)) {
+                System.out.println(new Ui().getReply(Command.FIND, input, arListTask));
+            } else {
+                new Ui().getReply(Command.INVALID, input, arListTask);
+            }
```

Comment:
Constantly reinitialising Ui might seem redundant. Maybe make Ui methods static? I think that might make your code look cleaner :)